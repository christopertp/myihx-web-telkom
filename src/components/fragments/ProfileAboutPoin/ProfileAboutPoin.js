import React, { useEffect } from 'react';
import WhiteCardPlain from '../../layouts/WhiteCardPlain';
import styles from './styles.scoped.css';
import { useDispatch, useSelector } from 'react-redux';

export default function ProfileAboutPoin() {

  const dispatch = useDispatch();
  const { aboutPoint, fetchAboutPoint } = useSelector(s => s.profile);

  useEffect(() => {
    dispatch(fetchAboutPoint());
  }, []);

  return(
    <WhiteCardPlain size="medium" subtitle={aboutPoint.pageDesc} title={aboutPoint.pageTitle}>
      <section className={styles.root}>
        {aboutPoint.content && aboutPoint.content.map((v, i) => (
          <section className={styles.contentAbout} key={i}>
            <strong>{v.header}</strong>
            {v.body.map((b, iB) => (
              <div key={iB}>
                <p>{b.text}</p>
                <section className={styles.howToGet}>
                  {b.descriptions && b.descriptions.map((d, iD) => (
                    <span key={iD}>
                      <strong>{d.header}</strong>
                      <small>{d.detail}</small>
                    </span>
                  ))}
                </section>
              </div>
            ))}
          </section>
        ))}
      </section>
    </WhiteCardPlain>
  );
}
