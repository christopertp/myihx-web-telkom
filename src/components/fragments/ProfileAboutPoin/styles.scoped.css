.contentAbout {
  margin-top: 1rem;
}

.contentAbout > div > p {
  font-size: 1rem;
  color: #333435;
}

.howToGet {
  display: inline-grid;
  grid-template-columns: 1fr 1fr;
  margin-top: 1rem;
}

.howToGet > span {
  border: 0.063rem solid #e5e6e6;
  border-radius: 0.5rem;
  display: inline-grid;
  margin: 0.5rem;
  padding: 0.75rem;
}

.howToGet > span > small {
  color: #333333;
}

.root {
  display: grid;
}

.root > small {
  font-size: 1rem;
  color: #333435;
}

.root > strong {
  font-size: 1.3rem;
}

@media only screen and (max-width: 48em) {
  .howToGet { margin-top: 0.5rem; }
  .howToGet > span { padding: 0.3rem; }
  .howToGet > span > strong { font-size: 0.8rem; }
  .howToGet > span > small { font-size: 0.7rem; }
  .root { margin-top: 6rem; }
  .root > small { margin-top: 0.3rem; }
}
