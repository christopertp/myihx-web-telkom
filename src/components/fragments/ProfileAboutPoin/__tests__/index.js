import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import ShallowRenderer from 'react-test-renderer/shallow';
import ProfileAboutPoin from '../ProfileAboutPoin';

useSelector.mockImplementation(fn => {
  fn({});
  return {
    fetchAboutPoint: jest.fn(),
    aboutPoint: {
      content: [{
        body: [{
          descriptions: [{
            detail: 'Bill payment amount for Dual Play customers',
            header: '10 poin / Rp 10.000',
          }],
          text: 'Poin myIndiHome is a loyalty program aimed for IndiHome Triple Play and Dual Play Package customers.'
        }],
        header: 'What is Poin myIndiHome?',
      }],
      pageDesc: 'Learn more about points and rewards from myIndiHome!',
      pageTitle: 'What is Poin myIndiHome?',
    }
  };
});

describe('src/components/fragments/ProfileAboutPoin', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<ProfileAboutPoin />);
    expect(tree).toMatchSnapshot();
  });

  test('call get about poin', () => {
    const dispatch = jest.fn();
    useDispatch.mockImplementationOnce(() => dispatch);

    ProfileAboutPoin();
    expect(dispatch).toHaveBeenCalled();
  });
});
