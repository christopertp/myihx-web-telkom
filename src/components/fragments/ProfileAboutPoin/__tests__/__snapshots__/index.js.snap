// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/fragments/ProfileAboutPoin render 1`] = `
<WhiteCardPlain
  back=""
  close=""
  help={false}
  icon=""
  size="medium"
  subtitle="Learn more about points and rewards from myIndiHome!"
  title="What is Poin myIndiHome?"
>
  <section>
    <section>
      <strong>
        What is Poin myIndiHome?
      </strong>
      <div>
        <p>
          Poin myIndiHome is a loyalty program aimed for IndiHome Triple Play and Dual Play Package customers.
        </p>
        <section>
          <span>
            <strong>
              10 poin / Rp 10.000
            </strong>
            <small>
              Bill payment amount for Dual Play customers
            </small>
          </span>
        </section>
      </div>
    </section>
  </section>
</WhiteCardPlain>
`;
